client/my-sites/email/email-non-domain-owner-message/index.tsx

```diff
@@ -0,0 +1,139 @@
+import { SiteDetails } from '@automattic/data-stores';
+import { useTranslate } from 'i18n-calypso';
+import { stringify } from 'qs';
+import { useDispatch } from 'react-redux';
+import { useDomainOwnerUserName } from 'calypso/components/data/query-domain-owner-username';
+import PromoCard from 'calypso/components/promo-section/promo-card';
+import { ResponseDomain } from 'calypso/lib/domains/types';
+import { emailManagementPurchaseNewEmailAccount } from 'calypso/my-sites/email/paths';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+
+import './style.scss';
+
+type EmailNonDomainOwnerMessageProps = {
+	domain?: ResponseDomain;
+	selectedSite?: SiteDetails | null;
+	usePromoCard: boolean;
+	source: 'email-comparison' | 'email-management';
+};
+
+export const EmailNonDomainOwnerMessage = ( props: EmailNonDomainOwnerMessageProps ) => {
+	const { domain, selectedSite, source, usePromoCard } = props;
+
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+
+	const ownerUserName = useDomainOwnerUserName( selectedSite, domain );
+
+	const isPrivacyAvailable = domain?.privacyAvailable;
+
+	const buildQueryString = ( parameters = {} ) =>
+		parameters ? stringify( parameters, { addQueryPrefix: true, skipNulls: true } ) : '';
+
+	const buildLoginUrl = () => {
+		const redirectUrlParameter = encodeURIComponent(
+			emailManagementPurchaseNewEmailAccount(
+				selectedSite?.slug ?? '',
+				domain?.name ?? '',
+				null,
+				'login-redirect'
+			)
+		);
+
+		return `/log-in/${ buildQueryString( {
+			email_address: ownerUserName,
+			redirect_to: redirectUrlParameter,
+		} ) }`;
+	};
+
+	const contactOwnerUrl = `https://privatewho.is/${ buildQueryString( {
+		s: domain?.name,
+	} ) }`;
+
+	const loginUrl = buildLoginUrl();
+
+	const onClickLink = ( eventType: 'owner_contact' | 'user_login' | 'contact_support' ) => {
+		const properties = {
+			owner_login: ownerUserName,
+		};
+
+		if ( eventType === 'owner_contact' ) {
+			dispatch( recordTracksEvent( `calypso_email_providers_owner_contact_click`, properties ) );
+
+			return;
+		} else if ( eventType === 'contact_support' ) {
+			dispatch( recordTracksEvent( `calypso_email_providers_contact_support_click`, properties ) );
+
+			return;
+		}
+
+		dispatch( recordTracksEvent( `calypso_email_providers_user_login_click`, properties ) );
+	};
+
+	const translateOptions = {
+		components: {
+			loginLink: (
+				<a
+					rel="noopener noreferrer"
+					href={ loginUrl }
+					onClick={ () => onClickLink( 'user_login' ) }
```

Comment:
Looks like we're still having issues with this link. When I tested in Firefox, the address in the address bar updated to the login link, but the login page did not load. When I added `target="_blank"` the login page opened in a new tab as expected.

<img width="1377" alt="Screen Shot 2022-09-27 at 8 07 28 PM" src="https://user-images.githubusercontent.com/104910361/192658802-396fd979-13f2-449d-b546-24fc5fc5b9dd.png">

Reply:
Weird issue!

Reply:
I found a workaround (used in other places of calypso. I think it's better to keep the same tab instead of adding a new tab.

Reply:
Could be something to do with `/log-in` not being defined as a section in `client/sections.js`? Maybe try with `rel="external"` on the link to force a non-push based navigation.